<launch>
    <!-- Select connecting device and serial port ，选择连接设备及串口-->
    <arg name="port" default="/dev/ttyTHS1" />
    <arg name="baud" default="1000000" />
    <!-- Load URDF, rviz, etc. on the parameter server,加载参数服务器上的URDF、rviz等 -->
    <arg name="model" default="$(find mycobot_description)/urdf/280jn/mycobot_urdf.urdf"/>
    <arg name="rvizconfig" default="$(find mycobot_280)/config/mycobot_with_marker.rviz" />
    <arg name="gui" default="false" />

    <arg name="num" default="0" />

    <param name="robot_description" command="$(find xacro)/xacro --inorder $(arg model)" />

    <!-- Combinejoin values to TF ，将值合并到TF-->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
    <!-- Show in Rviz，显示在Rviz -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />

    <!-- mycobot-topics， mycobot-话题-->
    <include file="$(find mycobot_communication)/launch/communication_topic.launch">
        <arg name="port" value="$(arg port)" /> 
        <arg name="baud" value="$(arg baud)" /> 
    </include>


    <!-- listen and pub the real angles ，监听并发布真实的角度-->
    <node name="real_listener" pkg="mycobot_280jn" type="listen_real_of_topic.py" />


    <!-- start 启动相机并识别aruco -->
    <include file="$(find realsense2_camera)/launch/rs_camera.launch">
        <arg name="color_width" value="680" />
        <arg name="color_height" value="480" />
        <arg name="enable_depth" value="false" />
     </include>

    <!--
    <arg name="markerId"        default="323"/>
    <arg name="markerSize"      default="0.03"/>    
    <arg name="marker_frame"    default="aruco_marker_frame"/>
    <arg name="ref_frame"       default=""/>  
    <arg name="corner_refinement" default="LINES" /> 
    
    <node pkg="aruco_ros" type="single" name="aruco_single">
        <remap from="/camera_info" to="/camera/color/camera_info" />
        <remap from="/image" to="/camera/color/image_raw" />
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="marker_id"          value="$(arg markerId)"/>
        <param name="reference_frame"    value="$(arg ref_frame)"/> 
        <param name="camera_frame"       value="camera_link"/>
        <param name="marker_frame"       value="$(arg marker_frame)" />
        <param name="corner_refinement"  value="$(arg corner_refinement)" />
    </node> 
    
    <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
        <remap from="image" to="/aruco_single/result"/>
        <param name="autosize" value="true" />
    </node>
    -->
    <!-- end 启动相机并识别aruco -->

    <!-- vision node -->
    <node name="realsense_detect_aruco" pkg="mycobot_280jn" type="realsense_detect_aruco.py" /> 
    <node name="publish_camera_frame" pkg="mycobot_280jn" type="publish_camera_frame.py" /> 
    <node name="publish_aruco_frame" pkg="mycobot_280jn" type="publish_aruco_frame.py" /> 


</launch>


    

